### PR TITLE
Add OSSGadget version 0.1.312

### PR DESCRIPTION
**Pull request by bykenx**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x ] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/ossgadget.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  url "https://github.com/microsoft/OSSGadget/releases/download/v#{version}/OSSGadget_macos_#{version}.tar.gz"
+  name "OSSGadget"
+  desc "Collection of tools for analyzing open source packages"
+  homepage "https://github.com/microsoft/OSSGadget"
+
+  livecheck do
+    url "https://github.com/microsoft/OSSGadget"
+    strategy :github_latest
+  end
```
Comment (neutric) [inline Casks/ossgadget.rb:13]: `strategy :github_latest` comes at the cost of a limited API access and is only necessary if there are also pre-releases. If all releases are production-ready, which it looks like to me, then you can omit the whole `livecheck` stanza and rely on the built-in `livecheck` for GitHub repos.

File: Casks/ossgadget.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  binary "#{staged_path}/OSSGadget_macos_#{version}/oss-detect-backdoor"
+  binary "#{staged_path}/OSSGadget_macos_#{version}/oss-detect-cryptography"
+  binary "#{staged_path}/OSSGadget_macos_#{version}/oss-diff"
+  binary "#{staged_path}/OSSGadget_macos_#{version}/oss-download"
+  binary "#{staged_path}/OSSGadget_macos_#{version}/oss-find-domain-squats"
+  binary "#{staged_path}/OSSGadget_macos_#{version}/oss-find-source"
+  binary "#{staged_path}/OSSGadget_macos_#{version}/oss-find-squats"
+  binary "#{staged_path}/OSSGadget_macos_#{version}/oss-health"
+  binary "#{staged_path}/OSSGadget_macos_#{version}/oss-metadata"
+  binary "#{staged_path}/OSSGadget_macos_#{version}/oss-risk-calculator"
```
Comment (neutric) [inline Casks/ossgadget.rb:26]: This PR is most likely going to be [rejected](https://docs.brew.sh/Acceptable-Casks#rejected-casks):

> The app is both open-source and CLI-only (i.e. it only uses the binary artifact). In that case, and in the spirit of deduplication, submit it first to Homebrew/core as a formula that builds from source. If it is rejected, you may then try again as a cask (link us to the issue so we can see the discussion and reasoning for rejection).

**Review by neutric — COMMENTED** [review]
Thanks for your first contribution to `homebrew-cask`, @bykenx! My review is in the code. Take it with a grain of salt as I am not a maintainer.

File: Casks/ossgadget.rb (hunk 2 above)
Comment (bykenx) [reply Casks/ossgadget.rb:26]: OK, I will try to submit to homebrew/core later.

Comment (neutric) [reply Casks/ossgadget.rb:26]: Wait and see what the maintainers say, maybe I overlooked something.

**Comment by gdams** [comment]
Thanks for your contribution @bykenx. To echo @neutric's point, this can likely be submitted to homebrew/core. Please let me know if you need assistance

**Comment by miccal** [comment]
Thank you @bykenx, but as pointed out by @neutric and @gdam and from our [acceptable Casks documentation](https://docs.brew.sh/Acceptable-Casks#rejected-casks):

> The app is both open-source and CLI-only (i.e. it only uses the `binary` artifact). In that case, and [in the spirit of deduplication](https://github.com/Homebrew/homebrew-cask/issues/15603), submit it first to [Homebrew/core](https://github.com/Homebrew/homebrew-core) as a formula that builds from source. If it is rejected, you may then try again as a cask (link us to the issue so we can see the discussion and reasoning for rejection).

**Comment by bykenx** [comment]
> Thanks for your contribution @bykenx. To echo @neutric's point, this can likely be submitted to homebrew/core. Please let me know if you need assistance

The app dependent on dotnet 6.x for packaging, it seems that the latest version of dotnet on homebrew/core is 5.x.
if dotnet does not release a new version, this app will not be published to homebrew/core

**Comment by miccal** [comment]
Ping @carlocab for your thoughts again, please.

**Comment by bykenx** [comment]
> Ping @carlocab for your thoughts again, please.

OK.

**Pull request closed without merging**